lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp

```diff
@@ -502,15 +534,24 @@ LLDBMemoryReader::getFileAddressAndModuleForTaggedAddress(
   }
 
   ModuleSP module = pair_iterator->second;
+  auto *section_list = module->GetSectionList();
+  if (section_list->GetSize() == 0) {
+    LLDB_LOG(log,
+             "[MemoryReader] Module with empty section list.");
+    return {};
+  }
   uint64_t file_address;
   if (pair_iterator == m_range_module_map.begin())
     // Since this is the first registered module,
     // clearing the tag bit will give the virtual file address.
     file_address = tagged_address & ~LLDB_FILE_ADDRESS_BIT;
   else
     // The end of the previous section is the start of the current one.
+    // We also need to add the first section's file address since we remove it
+    // when constructing the range to module map.
     file_address = tagged_address - std::prev(pair_iterator)->first;
 
+  file_address += section_list->GetSectionAtIndex(0)->GetFileAddress();
```

Comment:
This wasn't there before, should we comment why we're adding the starting address of [0]?